product/util.py

```diff
@@ -344,9 +345,9 @@ def _effective_images(
 
 def _enumerate_effective_images(
     component_descriptor: ComponentDescriptor,
-    image_reference_filter=lambda _: True,
+    image_reference_filters=(model.filter.MatchAllFilter(),),
```

Comment:
Another Approach: Define CompositeFilter class and use, instead of an iterable of filters.
Not sure whether it's better in the end